### ISSUE TITLE
Clarification on Ablation Study: "Replacing the fusion structure with a linear layer"

### ISSUE THREAD

**Issue by zk1009** [xyz189411yt/C4MMD#13]
Hi,

I am currently reading your paper and am very interested in the ablation experiments detailed, particularly in Table 2 .It's mentioned that: "Replacing the fusion structure in the model with a linear layer resulted in a significant decrease in performance."

I would greatly appreciate it if you could clarify what "Replacing the fusion structure in the model with a linear layer" specifically entails. Does this mean that the model, in this configuration, exclusively uses features from the text modality (e.g., feeding only text features into a linear layer for classification)? Or does it refer to a simpler method of multi-modal feature fusion (while still being multi-modal)?

Thanks!

**Comment by xyz189411yt** [comment]
Hi\~
Thanks for your kind words and support for our work!

In the ablation study, *replacing the fusion structure in the model with a linear layer* means that we first align the dimensions of the image and text vectors, then simply concatenate them using `torch.cat`. The resulting fused vector has the shape `[batch_size, img_vec_len + text_len, hidden_state]`.
Next, we apply `torch.mean` over the second dimension to get the average, and then use a linear layer to map the `hidden_state` to 2 dimensions. This gives us a final vector of shape `[batch_size, 1 (we remove this dimension using torch.squeeze), 2]`. Finally, we apply softmax or sigmoid for classification.

To sum up, I hope this answers your question. What we mean by *replacing the fusion structure in the model with a linear layer* is that we only replace the **Multimodal Fusion** module, keeping the rest of the model unchanged. This helps us demonstrate the necessity and effectiveness of the original fusion structure.
